sql/malsync/malsync.go

```diff
@@ -9,29 +9,33 @@ import (
 
 func GetSyncState(db sql.Executor) (time.Time, error) {
 	var timestamp time.Time
-	rows, err := db.Exec("select timestamp from malfeasance_sync_state",
+	rows, err := db.Exec("select timestamp from malfeasance_sync_state where id = 1",
 		nil, func(stmt *sql.Statement) bool {
 			v := stmt.ColumnInt64(0)
 			if v > 0 {
 				timestamp = time.Unix(v, 0)
 			}
 			return true
 		})
-	if err != nil {
+	switch {
+	case err != nil:
 		return time.Time{}, fmt.Errorf("error getting malfeasance sync state: %w", err)
-	} else if rows != 1 {
+	case rows <= 1:
+		return timestamp, nil
+	default:
 		return time.Time{}, fmt.Errorf("expected malfeasance_sync_state to have 1 row but got %d rows", rows)
 	}
-	return timestamp, nil
 }
 
 func updateSyncState(db sql.Executor, ts int64) error {
-	_, err := db.Exec("update malfeasance_sync_state set timestamp = ?1",
+	if _, err := db.Exec(
+		`insert into malfeasance_sync_state (id, timestamp) values(1, ?1)
+                   on conflict (id) do update set timestamp = ?1`,
```

Comment:
The migration that added `malfeasance_sync_state` was a bit problematic for the new schema handling as it did insert a single row into the table. With the new schema handling, the fresh database is initialized from the schema file which is maintained by means of `go generate` and contains the schema captured from the SQLite database. The captured schema cannot contain any data, and thus it's no longer possible to depend on the fact that `malfeasance_sync_state` is not empty. Thus, I've added a dummy ID to this table to serve as a primary key, and instead of relying on the single row to be always present I use this "upsert" construct that creates the row when it doesn't exist, which will be the case with the fresh db.